force-app/main/default/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -975,80 +975,65 @@ export default class GeFormRenderer extends LightningElement{
-        ];
-        // get label and value using apiName as key from fields for each section
-        let miniFieldWrapper = {};
-        sectionsList.forEach(section => {
-            miniFieldWrapper = { ...miniFieldWrapper, ...(section.getFieldValueAndLabel(DONATION_VALUES))};
-        });
-
         // if no donation donor selection, nothing to validate here yet
-        if ( isEmpty(miniFieldWrapper[DONATION_DONOR_FIELDS.donationDonorField].value) ) {
+        if (isEmpty(this.getFieldValueFromFormState(DONATION_DONOR_FIELDS.donationDonorField)) ) {
+            sectionsList.forEach( section => {
+                section.setCustomValidityOnFields(Object.values(DONATION_DONOR_FIELDS), '');
+            });
             return false;
         }
 
         // returns true when error message was generated
-        return this.getDonorTypeValidationError( miniFieldWrapper, sectionsList );
-    }
-
-    /**
-     * helper class for isDonorTypeInvalid, contains majority of logic
-     * @param fieldWrapper - Array, field ui-label and value using field-api-name as key
-     * @param sectionsList - Array, all sections
-     * @returns {boolean} - true if error message was generated, false if otherwise
-     */
-    getDonorTypeValidationError(fieldWrapper, sectionsList) {
-
         // get data import record helper
-        const di_record = this.getDataImportHelper(fieldWrapper);
+        const dataImportHelper = this.getDataImportHelper();
 
+        const isAccountDonor = dataImportHelper.donationDonorValue === DONATION_DONOR.isAccount1;
+        const areEmptyAccountFields = dataImportHelper.isAccount1ImportedEmpty && dataImportHelper.isAccount1NameEmpty;
+        const isContactDonor = dataImportHelper.donationDonorValue === DONATION_DONOR.isContact1;
+        const areEmptyContactFields = dataImportHelper.isContact1ImportedEmpty && dataImportHelper.isContact1LastNameEmpty;
         // donation donor validation depending on selection and field presence
-        let isError = (di_record.donationDonorValue === DONATION_DONOR.isAccount1) ?
-            di_record.isAccount1ImportedEmpty && di_record.isAccount1NameEmpty :
-            di_record.donationDonorValue === DONATION_DONOR.isContact1 &&
-            di_record.isContact1ImportedEmpty && di_record.isContact1LastNameEmpty;
+        const isError = isAccountDonor ? areEmptyAccountFields : isContactDonor && areEmptyContactFields;
 
         // process error notification when error
         if (isError) {
-            // highlight validation fields
-            this.highlightValidationErrorFields(di_record, sectionsList, ' ');
-            // set page error
-            this.hasPageLevelError = true;
-            this.pageLevelErrorMessageList = [ {
-                index: 0,
-                errorMessage: this.getDonationDonorErrorLabel(di_record, fieldWrapper)
-            } ];
+            this.displayDonorTypeError(dataImportHelper, sectionsList);
```

Comment:
I don't think we should call this function inside of `isDonorTypeInvalid()` since this is doing more than performing validation. I would think we would call this around line 950 in the `isFormValid()` function if `isDonorTypeInvalid()` returns true. Thoughts?

Reply:
I see what you meant originally now. If `this.isDonorTypeInvalid` returns false, then we can immediately apply the validation errors in the `isFormValid` function.